PluginAutotask/API/Utility/ApplyDynamicDate.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using Newtonsoft.Json;
+using PluginAutotask.DataContracts;
+
+namespace PluginAutotask.API.Utility
+{
+    public static partial class Utility
+    {
+        public static string DynamicDateConstant = "TODAYMINUS_";
+
+        public static Query ApplyDynamicDate(Query query)
+        {
+            if (!JsonConvert.SerializeObject(query.Filter).Contains(DynamicDateConstant))
+            {
+                return query;
+            }
+
+            foreach (var filter in query.Filter)
+            {
+                if (filter.Value is string)
+                {
+                    var value = (string)filter.Value;
+                    var parts = value.Split('_');
+                    if (parts.Length != 3)
```

Comment:
I do not see a check for the final element of the split string (`parts[2]`). As part of the unit test, a user could type in something other than DAYS as the final part of the constant (e.g., TODAYMINUS_N_**MINUTES**) and still have the unit test pass.

Reply:
Good point. I have added more conditions to fall into the error note.